crates/runtime/src/lib.rs

```diff
@@ -304,14 +333,18 @@ impl Runtime {
     }
 
     pub async fn start_pods_watcher(&mut self) -> notify::Result<()> {
-        let mut current_app = Arc::clone(&self.app);
-
         let mut rx = self.pods_watcher.watch()?;
 
         while let Some(new_app) = rx.recv().await {
+            let current_app = self.app.read().await;
```

Comment:
This code relies on the app state not changing between reading and writing so we should acquire the write lock here and then release it at the end of the loop iteration.

Reply:
The only concern here is that loading models/databases/other future functionality might want to use current app state. If we acquire write lock immediately, then it will result in a deadlock. As we have single thread/channel responsible for the state change then it should be fine not to lock the app immediately IMO.

Reply:
How would acquiring the lock immediately result in deadlock? Just acquire, read, write, then release the lock. I think it's very unlikely that we will continue to have a single thread changing state moving forward so relying on that isn't a good solution.

Reply:
I've just tested `while let Some(new_app) = rx.recv().await` for potential race condition with 5 sec async/await delay inside and its executed sequentially so we should be good for the potential case when app states are processed in parallel leading to broken state

Reply:
This particular thread won't race with itself but my concern is other threads changing the state. The reason we have the RwLock is to prevent this, so let's use it properly. Your code works on it's own for now, but it should work properly with other code reading/writing the App too.